cli/azd/pkg/apphost/manifest.go

```diff
@@ -8,12 +8,16 @@ import (
 	"path/filepath"
 	"strconv"
 
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
 	"github.com/azure/azure-dev/cli/azd/pkg/tools/dotnet"
+	"github.com/psanford/memfs"
 )
 
 type Manifest struct {
 	Schema    string               `json:"$schema"`
 	Resources map[string]*Resource `json:"resources"`
+	// BicepFiles holds any bicep files generated by Aspire next to the manifest file.
+	BicepFiles *memfs.FS `json:"-"`
```

Comment:
I would prefer to keep the manifest type something that feels like a model type for the aspire manifest - and not include this`BicepFiles` member (I'm fine if whatever produces the manifest starts instead return a tuple of Manifest and additional files.

But I'm also wondering why we have this at all - I suspect it is because when you go to emit the bicep you want to just dump all this stuff into disk - sort of figured however that we'd do this much later in the "compilation" pipeline instead of eagerly reading the files into memory and hanging on to their contents.  I didn't yet understand how you were consuming this property - so perhaps when I see the more complete end to end it will make sense.

Reply:
What about changing the name from `Manifest` to `AppHostPublishResult` or something more generic than what the manifest is?

The single json manifest as the output from `dotnet --publish ...` is evolving to be now a `manifest + generated files`

> But I'm also wondering why we have this at all

Took me a while to figure as well!!  hahaha.. 
AFAIK, the desing looks like:
- We have a `dotnet importer` on package: `project`. This importer knows uses the dotnet cli and these inputs:

![image](https://github.com/Azure/azure-dev/assets/24213737/7a60e2af-f4cc-4164-819c-c43b616dd444)

  and it provides the method:

```go
func (ai *DotNetImporter) readManifest(ctx context.Context, svcConfig *ServiceConfig) (*apphost.Manifest, error) {
```

   readManifest() provides a cache per `AppHost path` (while a command runs).  Which means that we should ever requiere to run `dotnet --publish ...` once per `azd command run`.  But in practice, it is hapening multiple times because the `ioc - container` produces one scoped `dotnet importer`:

```
container.MustRegisterScoped(project.NewDotNetImporter)
```

   **We could explore moving the dotnet importer as a singleton, to fully use the cache that it is offering.  Right now, the cache works only per `command action`, which means that `azd up` publishes the appHost 3 times (package, provision and deploy).

Ok, moving on...
Having a manifest, the next layer to look at is the `appHost` package, which exports a few functions that are consumed by the dotnet importer. For example `BicepTemplate()`, which takes the `publishResult (aka the manifest)` and produces the bicep infrastructure.  Such `bicep infrastructure` is represented by `*memfs.FS` (in memory filesystem).  Once it is produced, the dotnet importer writes the contents from the memfs to the disk :).

Moving one layer down into the appHost (which can be seen as the implementation details below the dotnet importer), we find the `infraGenerator`, which loads `dotnetPublishResult` and compiles it, which in simple world would be to fill typed-structures in memory to be used with go-templates to produce files.  The produced files were written in a new `*memfs.FS` and returned to the `dotnet importer` (as mentioned before).  AND here is where I connected the new changes:
Instead of having the appHost creating a new `*memfs.FS`, the `dotnet importer` adds the `*memfs.FS` within the PublishResult to be used by the appHost to write the infra entry point (main.bicep, params, resources, etc).

Reply:
I think I understand why we have this now - the problem is that we have to create a temporary directory to write the manifest to - which we then parse.  That directory also contains some bicep files and we need to retain them with the manifest in memory as we have done here because we are going to lose the files on disc shortly, because we'll clean up the temporary folder.

I had thought that perhaps we would be able to reference these files when we needed to use them (e.g. copy them to their final location later in the pipeline from the location the manifest writer wrote them to) but they will no longer exist when we try to use them, because we cleaned up that temp folder.

Reply:
Happy with the design now - thanks for walking me through it :-)